From 2444e6668658d7577cb9cb49062f3957bb44564a Mon Sep 17 00:00:00 2001
From: Hariprasad Kelam <hkelam@marvell.com>
Date: Fri, 11 Jun 2021 20:09:00 +0530
Subject: [PATCH 23/23] octeontx2-af: Read mac address from correct offset

commit a6d21d9c0883780909ff23c8f941ae5093ddd5f2 from
git@git.assembla.com:cavium/WindRiver.linux.git

DMAC CAM is per CGX/RPM block dont pass lmac_id while
accessing the same.

Fixes: adf334674080 ("octeontx2-af: add multicast filter support")

Change-Id: Ibc262b6200f58a94c8eda05ebfa1778a76b5ecfb
Signed-off-by: Hariprasad Kelam <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/53995
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 429407d2d4dc..7aec416c2bc5 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -357,8 +357,7 @@ int cgx_lmac_addr_del(u8 cgx_id, u8 lmac_id, u8 index)
 	index = lmac_id * lmac->mac_to_index_bmap.max + index;
 
 	/* Read MAC address to check whether it is ucast or mcast */
-	cfg = cgx_read(cgx_dev, lmac_id,
-		       (CGXX_CMRX_RX_DMAC_CAM0 + (index * 0x8)));
+	cfg = cgx_read(cgx_dev, 0, (CGXX_CMRX_RX_DMAC_CAM0 + (index * 0x8)));
 
 	cfg2mac(cfg, mac);
 	if (is_multicast_ether_addr(mac))
-- 
2.31.1

